tests/testserver/server.py

```diff
@@ -39,6 +39,10 @@ def __init__(self, handler, host='localhost', port=0, requests_to_handle=1, wait
         self.ready_event = threading.Event()
         self.stop_event = threading.Event()
 
+        self.server_sock = self._create_socket_and_bind()
```

Comment:
I'm *really* nervous about this being in `__init__`. Is there any reason it's here?